### PR TITLE
Optionally remove prepending of linode ID, added private networking support

### PR DESCRIPTION
**Pull request by bartmeuris**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

cloud/linode
##### ANSIBLE VERSION

```
ansible 2.1.0 (devel 54acdd7ead) last updated 2016/04/11 11:16:30 (GMT +000)
  lib/ansible/modules/core: (detached HEAD c52f475c64) last updated 2016/04/11 11:16:45 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD 5abb914315) last updated 2016/04/11 11:16:53 (GMT +000)
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

This PR merges a few pre-`ansible-modules-core` issues (see https://github.com/ansible/ansible/pull/7223), regarding prepending the ID to the name, which makes idempotent operation impossible. An instance is now also looked up by it's name if the new `name_add_id` option is set to `False`.

Some additional features were added, like the ability to select the separator used between the ID and the name you choose. By default, `_` is used, which is the same default as before.

There is now also a `private_networking` option which will add a static IP to an instance. Note that this will only be applied on creation.

The default behavior of the module remains the same, unless configured otherwise to maintain backward compatability.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @bartmeuris. To the current maintainers, @zbal please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/linode/linode.py [hunk 1]
```diff
@@ -34,9 +34,18 @@
   name:
     description:
      - Name to give the instance (alphanumeric, dashes, underscore)
-     - To keep sanity on the Linode Web Console, name is prepended with LinodeID_
     default: null
     type: string
+  name_add_id:
```
Comment (jjshoe) [inline cloud/linode/linode.py:39]: Missing version added

File: cloud/linode/linode.py [hunk 2]
```diff
@@ -34,9 +34,18 @@
...
+    description:
+    - prepend the linode ID to the name
+    default: True
+    type: bool
+  name_id_separator:
```
Comment (jjshoe) [inline cloud/linode/linode.py:45]: Missing version added

File: cloud/linode/linode.py [hunk 3]
```diff
@@ -79,6 +88,11 @@
      - datacenter to create an instance in (Linode Datacenter)
     default: null
     type: integer
+  private_networking:
```
Comment (jjshoe) [inline cloud/linode/linode.py:93]: Missing version added

**Comment by bartmeuris** [comment]
The travis build seem to fail on stuff I did not touch (something in the yum module?)

**Comment by qrkourier** [comment]
resurrecting this in https://github.com/ansible/ansible/pull/28947/

**Pull request closed without merging**